### PR TITLE
[Security][Profiler] Display the original expression in 'Access decision log'

### PR DESCRIPTION
**Pull request by lyrixx**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #24127
| License       | MIT
| Doc PR        |

---

before:

screenshot at 2018-02-22 18-22-28

after:

after

### CODE REVIEW

**Comment by stof** [comment]
I suggest *also* adding my patch from https://github.com/symfony/symfony/issues/24127#issuecomment-367317851 (the second one), to use `profiler_dump(first)` when `first` represents a non-string value, instead of displaying the string-casting of the Data object (the class name with the number of property in it is totally useless).

We might even consider this as a bugfix.

File: src/Symfony/Bundle/SecurityBundle/Resources/views/Collector/security.html.twig [hunk 1]
```diff
@@ -317,7 +317,12 @@
                         </td>
                         <td>
                             {% if decision.attributes|length == 1 %}
-                                {{ decision.attributes|first }}
+                                {% set first = decision.attributes|first %}
+                                {% if first.expression is defined %}
+                                    Expression: <code><pre>{{ first.expression }}</pre></code>
```
Comment (stof) [inline src/Symfony/Bundle/SecurityBundle/Resources/views/Collector/security.html.twig]: should be `<pre><code></code></pre>` as `<code>` is an inline element while `<pre>` is a block element.

**Review by stof — CHANGES_REQUESTED** [review]

**Comment by lyrixx** [comment]
@stof  I addressed your comments.

**Review by Simperfit — APPROVED** [review]

**Comment by fabpot** [comment]
Looks like a bug fix to me, no?

**Comment by lyrixx** [comment]
I have rebased my branch onto 3.4, and changed the base branch for this PR in order to merge it as a bug fix.

**Review by chalasr — APPROVED** [review]

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @lyrixx.

**Pull request merged**